execution/scan/vector_selector.go

```diff
@@ -165,35 +187,38 @@ func (o *vectorSelector) loadSeries(ctx context.Context) error {
 }
 
 // TODO(fpetkovski): Add max samples limit.
-func selectPoint(it *storage.MemoizedSeriesIterator, ts, lookbackDelta, offset int64) (int64, float64, *histogram.FloatHistogram, bool, error) {
+// To push down the timestamp function into this file and store the timestamp in the value for each series, you can modify the selectPoint function.
+func selectPoint(it *storage.MemoizedSeriesIterator, ts, lookbackDelta, offset int64) (point, bool, error) {
```

Comment:
My suggestion would be to roll back the changes and use a `bool` flag here to indicate whether to return the timestamp as the point value. Then for the `case chunkenc.ValFloat` case in the switch statement we can return `t` as the point value instead of `v`.

Reply:
Is it possible to return the timestamp as the point value in` case chunkenc.ValFloat `without the need to set a boolean flag for it? This would allow us to handle non-histogram data directly.

Reply:
Yes, the return types would remain as they are. The idea is to use a `bool` as an argument to the function in order to decide whether the `chunkenc.ValFloat` case would return `t` or `v`.

I guess we would need to do the same for histograms, based on this boolean we would either return the histogram itself or `t` as the `float64` return value.

Reply:
Okay I got the idea behind it !